src/src/hw9/hillel/ArrayValueCalculator.java

```diff
@@ -0,0 +1,40 @@
+package hw9.hillel;
+
+public class ArrayValueCalculator {
+
+    public static int doCalc(String[][] arr) throws ArraySizeException, ArrayDataException {
+        if (arr.length != 4 || arr[0].length != 4) {
```

Comment:
`arr.length != 4` <- this is correct. No questions.
But this -> `arr[0].length != 4` is not correct.
You've started in the correct way, but haven't finished the validation. You're validating only 1 row here, but it's required to validate them all because the matrix must be of size 4x4 according to the description.